flang/lib/Optimizer/Transforms/OMPFunctionFiltering.cpp

```diff
@@ -32,7 +33,15 @@ class OMPFunctionFilteringPass
 public:
   OMPFunctionFilteringPass() = default;
 
+  void getDependentDialects(DialectRegistry &registry) const override {
+    // fir::UndefOp creation requires that FIROpsDialect dialect is loaded.
+    registry.insert<fir::FIROpsDialect>();
+  }
+
   void runOnOperation() override {
+    MLIRContext *context = &getContext();
+    context->getOrLoadDialect<fir::FIROpsDialect>();
+    OpBuilder opBuilder(context);
```

Comment:
@skatrak @kiranchandramohan I had to add this change in order to create the `fir::UndefOp` when the patch is run standalone. Hope this is okay.

Reply:
Would adding in https://github.com/llvm/llvm-project/blob/dbd00c3b5d8a0703e1457b5575b3400e2ebc0157/flang/include/flang/Optimizer/Transforms/Passes.td#L333 work for you?

Reply:
Yes this works, thanks for the suggestion.